src/dateUtils.js

```diff
@@ -0,0 +1,44 @@
+const { DateTime, Duration } = require('luxon');
+
+const toDate = val => {
+  if (DateTime.isDateTime(val)){
+    return val;
+  } else if (typeof val === 'number'){
+    return DateTime.fromMillis(val);
+  } else if (typeof val === 'string'){
+    if (val.match(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/)){
+      return DateTime.fromISO(val);
+    } else if (val.match(/^[0-9]{2}\/[0-9]{2}\/[0-9]{4}$/)){
+      return DateTime.fromFormat(val, 'dd/MM/yyyy');
+    }
+    return DateTime.fromRFC2822(val);
+  } else if (typeof val === 'object'){
+    if (typeof val.getMonth === 'function'){
```

Comment:
is `val instanceof Date` more clear?

Reply:
Invalid dates are also instances of Date
```javascript
let t = new Date('notadate');
t instanceof Date; // returns true
```
I've discovered this check is still inadequate, a better combination is instanceof check and valueOf shouldn't be NaN

Reply:
Updated